scripts/gdb-pretty-printers.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import gdb
+
+
+if not 'ThrustVectorPrinter' in dir():
+    print("This file expects the Thrust pretty-printers to be loaded already. "
+          "Either load them manually, or use the generated gdb-pretty-printers.py "
+          "in the build directory")
+else:
+    class HostIterator(Iterator):
+        """Iterates over arrays in host memory"""
+
+        def __init__(self, start, size):
+            self.item = start
+            self.size = size
+            self.count = 0
+
+        def __iter__(self):
+            return self
+
+        def __next__(self):
+            if self.count >= self.size:
+                raise StopIteration
+            elt = self.item.dereference()
+            count = self.count
+            self.item = self.item + 1
+            self.count = self.count + 1
```

Comment:
... and similarly elsewhere.
```suggestion
            self.item += 1
            self.count += 1
```